### PR TITLE
chore: stablise CircleCI builds

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

*work-in-progress* Throwing stuff at Circle CI until something sticks ;) 

References: 
* https://stackoverflow.com/questions/44046366/increase-javascript-heap-size-in-create-react-app-project

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -344,7 +344,7 @@ workflows:
           matrix:
             parameters:
               BUILD_ENV:
-                ['FORCE_COLOR=1 REACT_APP_SITE_VARIANT=test-ci CI=false']
+                ['FORCE_COLOR=1 REACT_APP_SITE_VARIANT=test-ci']
       - test_e2e:
```
Comment (thisislawatts) [inline .circleci/config.yml:348]: We **are** in a CI environment, so let's opt into all of the less known behaviour that comes with that 
https://create-react-app.dev/docs/running-tests/#continuous-integration

Comment (chrismclarke) [reply .circleci/config.yml:348]: So the only reason for that previously was lint warnings are thrown as errors in CI, but with your PR to fix hopefully that should be fine now

**Comment by chrismclarke** [comment]
Just to capture what was mentioned in the slack thread, a couple issues of relevance (noted in package.json):
```
 "__NOTE2__": "2021-07-06 ram issues: https://github.com/facebook/create-react-app/issues/10154",
    "react-scripts/eslint-webpack-plugin": "2.4.1",
    "__NOTE3__": "2021-10-03 terser issues: https://github.com/facebook/create-react-app/issues/6334",
```

**Comment by chrismclarke** [comment]
If you need any extra permissions on CircleCI for investigating just let me know (I forget what people can/can't access when a member of the repo)

**Review by chrismclarke — APPROVED** [review]

**Pull request merged**